contracts/zero-ex/contracts/src/features/NativeOrdersFeature.sol

```diff
@@ -751,6 +907,48 @@ contract NativeOrdersFeature is
         orderInfo.status = LibNativeOrder.OrderStatus.FILLABLE;
     }
 
+    /// @dev Calculate the actual fillable taker token amount of an order
+    ///      based on maker allowance and balances.
+    function _getActualFillableTakerTokenAmount(
+        GetActualFillableTakerTokenAmountParams memory params
+    )
+        private
+        view
+        returns (uint128 actualFillableTakerTokenAmount)
+    {
+        if (params.orderMakerAmount == 0 || params.orderTakerAmount == 0) {
+            // Empty order.
+            return 0;
+        }
+        if (params.orderInfo.takerTokenFilledAmount >= params.orderTakerAmount) {
+            // Already fully filled.
```

Comment:
Should we check if the order status is `FILLABLE`? Could be cancelled or just invalid.

Reply:
Oh, doh, good catch.